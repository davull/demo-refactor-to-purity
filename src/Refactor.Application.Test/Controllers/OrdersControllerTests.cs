using System.Net;
using FluentAssertions;
using Snapshooter.NUnit;

namespace Refactor.Application.Test.Controllers;

public class OrdersControllerTests : IntegrationTestBase
{
    [Test]
    public async Task Should_Return_Ok()
    {
        var response = await GetAsync("/orders");
        response.StatusCode.Should().Be(HttpStatusCode.OK);
    }

    [Test]
    public async Task Should_Match_Snapshot()
    {
        var content = await GetStringAsync("/orders");
        content.Should().MatchSnapshot();
    }
}