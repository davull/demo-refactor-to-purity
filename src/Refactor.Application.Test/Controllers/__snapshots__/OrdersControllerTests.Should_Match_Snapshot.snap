[{"id":"687b4a03-6f3d-4f42-9ee6-53d5cf7e80d5","customer":{"id":"2f6c199e-be0f-4515-9e42-3e3f88c0523a","firstName":"Peter","lastName":"Pan","email":"peter.pan@example.com"},"items":[{"id":"e008de98-6f3e-4a3e-a81a-e094a923c5c3","productId":"7db60574-f667-46e9-941e-b775a2542e33","quantity":1,"grossPrice":11.90,"netPrice":10.00,"taxRate":19,"taxAmount":1.90,"totalGrossPrice":11.90,"totalNetPrice":10.00},{"id":"84dfe6a1-c537-4468-bb6d-4a93cba99ae8","productId":"cd6b6e1c-8cf2-4769-8205-39f41179c135","quantity":2,"grossPrice":54.145,"netPrice":45.50,"taxRate":19,"taxAmount":8.645,"totalGrossPrice":108.290,"totalNetPrice":91.00}],"orderDate":"2021-01-01T10:30:00"},{"id":"f1cf45fb-33b5-4a3d-b684-2e3a00a0ebea","customer":{"id":"2f6c199e-be0f-4515-9e42-3e3f88c0523a","firstName":"Peter","lastName":"Pan","email":"peter.pan@example.com"},"items":[{"id":"0f90b82b-0c29-4417-8f18-da20b734d3a7","productId":"ff08a855-9f82-4f2d-bf98-f0a929f4ef90","quantity":1,"grossPrice":11.662,"netPrice":9.80,"taxRate":19,"taxAmount":1.862,"totalGrossPrice":11.662,"totalNetPrice":9.80},{"id":"84319c54-7eab-416f-8172-bd472635dc88","productId":"2cfcd5d9-dcb6-495d-9878-8d8f6fe85fac","quantity":1,"grossPrice":12.495,"netPrice":10.50,"taxRate":19,"taxAmount":1.995,"totalGrossPrice":12.495,"totalNetPrice":10.50},{"id":"78263e1a-1023-494f-a89e-dbe24d8819de","productId":"89671827-5b8c-40cd-a533-af04224f5f29","quantity":4,"grossPrice":23.80,"netPrice":20.00,"taxRate":19,"taxAmount":3.80,"totalGrossPrice":95.20,"totalNetPrice":80.00},{"id":"3400dfa0-49ee-441c-8b93-a9040f6c83cd","productId":"d583263d-2169-4dc7-acbe-52846b79b7a9","quantity":2,"grossPrice":36.295,"netPrice":30.50,"taxRate":19,"taxAmount":5.795,"totalGrossPrice":72.590,"totalNetPrice":61.00}],"orderDate":"2021-01-11T12:30:00"},{"id":"6a72a608-77ea-4742-80b1-2152daf260c9","customer":{"id":"bfbffb19-cdd4-42ac-b536-606a16d03eae","firstName":"John","lastName":"Doe","email":"john.doe@example.com"},"items":[],"orderDate":"2021-01-20T20:45:00"}]
